CIPs/cip-0032.md

```diff
@@ -0,0 +1,117 @@
+
+If we observe that a validator is relying too heavily upon the `SignalOnline` functionality when they're receiving legitimate attestation requests we can add mitigations such as introducing an increasing financial penalty for subsquent signals or by decreasing the time that signal will last for (lowering `S`) per user.
+
+### Requiring Attestation Service
+
+**Note that the slasher expects each validator to be running an Attestation Service.** Unlike today, where it's optional for a Validator to run an attestation node, this proposal would make it mandatory. Having more attestation nodes in the pool improves the security of the phone number verification by making it more difficult to game the random selection of a set of colluding issuers. This proposal recommends providing at least three weeks from time of merge of this CIP to the time of enacting the slashing mechanism to give everyone an opportunity to set up their Attestation Service.
+
+### Nothing at stake problem
+
+Unlike other slashing contracts which use `SlasherUtil.performSlashing()` to both slash _and_ remove the validator, this `AttestationSlasher` would directly call `LockedGold.slash()`, only slashing stake and not removing the validator. This decreases the potential risk of having multiple validators removed simultaneously due to an overly aggressive `AttestationSlasher` policy. If the validator is slashed it may be that their locked stake drops below the `validatorLockedGoldRequirements` (currently 10K CELO). By dropping below `validatorLockedGoldRequirements` alone, the validator is not removed from the validator set but they will cease to earn epoch rewards. If they continue to get slashed by the `AttestationSlasher`, it may be that they drop to a near-zero stake while still remaining to be a validator. This is a dangerous situation, since this validator is performing an important role for the network but has nothing at stake. To resolve this tension, we'll introduce a new `MinStakeSlasher` which will force a validator to be de-affiliated if their balance drops below the `penalty + reward` threshold of any of the registered slasher contracts. Once a validator has been de-affiliated, they'll need to return their stake amount to the `validatorLockedGoldRequirements` requirement before being able to affiliate again.
```

Comment:
nit:  I *think* the implication that you are alluding here is that these validators will effectively be removed when going below `validatorLockedGoldRequirements` after the next epoch election.  It would be nice if that's explicitly stated.